src/main/java/me/i509/fabric/bulkyshulkies/extension/QuickShulkerHook.java

```diff
@@ -0,0 +1,81 @@
+
+      QuickOpenableRegistry.register(((player, stack) -> ContainerProviderRegistry.INSTANCE.openContainer(QS$SHULKER_SCROLLABLE_CONTAINER,player,
+              writer -> writer.writeInt(player.inventory.getSlotWithStack(stack)))),
+              ShulkerBlocks.COPPER_SHULKER_BOX, ShulkerBlocks.IRON_SHULKER_BOX, ShulkerBlocks.SILVER_SHULKER_BOX, ShulkerBlocks.SLAB_SHULKER_BOX);
+
+      QuickOpenableRegistry.register(((player, stack) -> ContainerProviderRegistry.INSTANCE.openContainer(QS$SHULKER_13x7_CONTAINER,player,
+              writer -> writer.writeInt(player.inventory.getSlotWithStack(stack)))),
+              ShulkerBlocks.OBSIDIAN_SHULKER_BOX, ShulkerBlocks.PLATINUM_SHULKER_BOX);
+
+      QuickOpenableRegistry.register(((player, stack) -> ContainerProviderRegistry.INSTANCE.openContainer(QS$SHULKER_11x7_CONTAINER,player,
+              writer -> writer.writeInt(player.inventory.getSlotWithStack(stack)))),
+              ShulkerBlocks.DIAMOND_SHULKER_BOX);
+
+      QuickOpenableRegistry.register(((player, stack) -> ContainerProviderRegistry.INSTANCE.openContainer(QS$SHULKER_9x7_CONTAINER,player,
+              writer -> writer.writeInt(player.inventory.getSlotWithStack(stack)))),
+              ShulkerBlocks.GOLD_SHULKER_BOX);
+   }
+   public static GenericContainer13x7 create13x7(int syncId, Identifier identifier, PlayerEntity player, PacketByteBuf buf) {
+      int invSlot = buf.readInt();
+      ItemStack stack = player.inventory.getInvStack(invSlot);
+      return  new GenericContainer13x7(syncId, ShulkerBoxSlot::new, player.inventory, new ItemStackInventory(stack, getSlotCount(stack)), new TranslatableText("container.shulkerBox"));
+   }
+   public static GenericContainer11x7 create11x7(int syncId, Identifier identifier, PlayerEntity player, PacketByteBuf buf) {
+      int invSlot = buf.readInt();
+      ItemStack stack = player.inventory.getInvStack(invSlot);
+      return  new GenericContainer11x7(syncId, ShulkerBoxSlot::new, player.inventory, new ItemStackInventory(stack, getSlotCount(stack)), new TranslatableText("container.shulkerBox"));
+   }
+   public static GenericContainer9x7 create9x7(int syncId, Identifier identifier, PlayerEntity player, PacketByteBuf buf) {
+      int invSlot = buf.readInt();
+      ItemStack stack = player.inventory.getInvStack(invSlot);
+      return  new GenericContainer9x7(syncId, ShulkerBoxSlot::new, player.inventory, new ItemStackInventory(stack, getSlotCount(stack)), new TranslatableText("container.shulkerBox"));
+   }
+   public static ScrollableContainer createScrollable(int syncId, Identifier identifier, PlayerEntity player, PacketByteBuf buf) {
+      int invSlot = buf.readInt();
+      ItemStack stack = player.inventory.getInvStack(invSlot);
+      return  new ScrollableContainer(syncId, ShulkerBoxSlot::new, player.inventory, new ItemStackInventory(stack, getSlotCount(stack)), new TranslatableText("container.shulkerBox"));
+   }
+   public static int getSlotCount(ItemStack stack) {
```

Comment:
I would reccomend instanceof checking the item before returning a slot count. If the item in the stack doesn't match a shulker box just return 0 or an OptionalInt

Reply:
Quick shulker should never give an item that isn’t registered, but it doesn’t hurt to be safe